subxt/src/metadata/metadata_type.rs

```diff
@@ -103,6 +103,15 @@ impl Metadata {
             .ok_or(MetadataError::PalletNotFound)
     }
 
+    /// Returns a reference to [`PalletMetadata`].
+    pub fn pallet_by_index(&self, pallet_index: u8) -> Result<&PalletMetadata, MetadataError> {
+        self.inner
+            .pallets
+            .values()
+            .find(|pallet_metadata| pallet_metadata.index == pallet_index)
+            .ok_or(MetadataError::PalletNotFound)
+    }
```

Comment:
Maybe won't be needed when moving to using names over indexes

Reply:
You're right, I removed the function.